src/visitors/rules/semantic/operation-tags-defined.js

```diff
@@ -0,0 +1,76 @@
+/* eslint-disable no-underscore-dangle */
+class OperationTags {
+  static get rule() {
+    return 'operation-tags-defined';
+  }
+
+  constructor() {
+    this.globalTagNames = [];
+    this.operationTags = []; // { name, path }
+  }
+
+  _finishValidation() {
+    return {
+      onExit: (_node, _, ctx) => {
+        const errors = [];
+        for (const tag of this.operationTags) {
+          if (this.globalTagNames.indexOf(tag.name) === -1) {
+            const tmpPath = ctx.path;
+            ctx.path = tag.path;
```

Comment:
I believe you may lose `file` context here.

Could you modify code to create errors immediately vs postpone it Root exit?

Reply:
👍, Updated to create errors in-place